### ISSUE TITLE
mgmt, condition in RBAC (ABAC support)

### ISSUE THREAD

**Issue by weidongxu-microsoft** [Azure/azure-sdk-for-java#25815]
https://docs.microsoft.com/en-us/azure/role-based-access-control/conditions-overview

Need upgrade to e.g. package-2020-10-01-preview tag in authorization.

**Comment by weidongxu-microsoft** [comment]
blocking issue is a bug in codegen for `package-2020-10-01-preview`.

Please try `package-2020-08-01-preview` first.

PS: 2020-10-01 appears missing following 2 JSONs that available in previous api-versions.
```
- Microsoft.Authorization/preview/2015-06-01/authorization-ClassicAdminCalls.json
- Microsoft.Authorization/stable/2015-07-01/authorization-ElevateAccessCalls.json
```

**Comment by weidongxu-microsoft** [comment]
Manual code part, likely only need to expose the condition (https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-08-01-preview/authorization-RoleAssignmentsCalls.json#L677-L680).

**Comment by XiaofeiCao** [comment]
@weidongxu-microsoft 
codegen from `package-2020-08-01-preview` seems to be missing `withScope()` method from `RoleAssignmentInner`, while `package-2020-03-01-preview` does not... I tried to compare swagger definitions of `RoleAssignment` in both versions and failed to notice anything different.
* 2020-08-01-preview
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-08-01-preview/authorization-RoleAssignmentsCalls.json#L719-L745

* 2020-03-01-preview
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-03-01-preview/authorization-RoleAssignmentsCalls.json#L604-L630

Maybe just use 2020-03-01-preview for now?

**Comment by weidongxu-microsoft** [comment]
scope is readonly in 2020-08-01
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-08-01-preview/authorization-RoleAssignmentsCalls.json#L645-L649

It seems OK to be readonly?

**Comment by XiaofeiCao** [comment]
With the application and service principal permission been assigned, I was able to run the tests under authorization. However, I’ve noticed some application tests are broken:
1. Current version `2020-08-01` leaves `filter` parameter unencoded by `"x-ms-skip-url-encoding": true`, which leads to error in query, e.g. `xxx&$filter=principalId eq '50e49d4b-ad87-4218-b752-e5630872c50a'`
`2020-08-01`
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-08-01-preview/authorization-RoleAssignmentsCalls.json#L768
`2018-01-01`
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-01-01-preview/authorization-RoleAssignmentsCalls.json#L509-L514
I added encoding to the filter parameter to solve the issue.
2. I failed to run the application test even with previous version. 
code snippet: 
```java
application =
       authorizationManager
          .applications()
          .define(name)
          .withSignOnUrl("http://easycreate.azure.com/" + name)
          .definePasswordCredential("passwd")
          .withDuration(Duration.ofDays(700))
          .attach()
          .defineCertificateCredential("cert")
          .withAsymmetricX509Certificate()
          .withPublicKey(readAllBytes(this.getClass().getResourceAsStream("/myTest.cer"))) //accepts PEM without `BEGIN CERTIFICATE`
          .withDuration(Duration.ofDays(100))
          .attach()
          .create();
```
The problem lies in service side. It accepts PEM without ```-----BEGIN CERTIFICATE-----``` surrounding and rejects PEM with it. Do we need to strip the surrounding for the user to prevent this breaking change?
